src/net/connection/virtual_connection.rs

```diff
@@ -41,6 +53,62 @@ impl VirtualConnection {
         let now = Instant::now();
         now.duration_since(self.last_heard)
     }
+
+    /// This prepossess the given buffer to be send over the network.
+    /// 1. It will append the right header.
+    /// 2. It will perform some actions related to how the packet should be delivered.
+    pub fn process_outgoing(&mut self, payload: &[u8], delivery_method: DeliveryMethod) -> NetworkResult<PacketData> {
+        let channel: &mut Channel = match delivery_method {
+            DeliveryMethod::Unreliable => { &mut self.unreliable_channel },
+            DeliveryMethod::ReliableUnordered => { &mut self.reliable_unordered_channel },
+            DeliveryMethod::ReliableOrdered => { &mut self.reliable_ordered_channel },
+            DeliveryMethod::SequencedOrdered => { &mut self.sequenced_ordered_channel },
+            DeliveryMethod::SequencedUnordered => { &mut self.sequenced_unordered_channel },
+        };
+
+        let packet_data: PacketData = channel.pre_process(payload, delivery_method)?;
+
+        Ok(packet_data)
+    }
+
+    /// This process the incoming data and returns an packet if the data is complete.
+    ///
+    /// Returns `Ok(None)`
+    /// 1. In case if of fragmentation and not all fragments are received.
```

Comment:
Should be `In the case of fragmentation and not all fragments are received`. This is an incomplete sentence though, not sure what you mean.